sw/airborne/filters/median_filter.h

```diff
@@ -75,61 +76,175 @@ inline int32_t update_median_filter(struct MedianFilterInt *filter, int32_t new_
     }
     filter->sortData[j + 1] = temp;
   }
-  return filter->sortData[(MEDIAN_DATASIZE) >> 1]; // return data value in middle of sorted array
+  // return data value in middle of sorted array
+  return get_median_filter_i(filter);
 }
 
-inline int32_t get_median_filter(struct MedianFilterInt *filter)
+inline int32_t get_median_filter_i(struct MedianFilterInt *filter)
 {
-  return filter->sortData[(MEDIAN_DATASIZE) >> 1];
+  if (filter->size % 2 == 0){
+    return (filter->sortData[filter->size / 2] + filter->sortData[filter->size / 2 - 1]) / 2;
+  } else {
+    return filter->sortData[filter->size >> 1];
+  }
 }
 
 struct MedianFilter3Int {
   struct MedianFilterInt mf[3];
 };
 
-#define InitMedianFilterVect3Int(_f) {  \
-    for (int i = 0; i < 3; i++) {         \
-      init_median_filter(&(_f.mf[i]));    \
-    }                                     \
+#define InitMedianFilterVect3Int(_f) {                        \
+    for (int i = 0; i < 3; i++) {                             \
+      init_median_filter_i(&(_f.mf[i]), MEDIAN_DEFAULT_SIZE); \
```

Comment:
why not passing the size parameters in this macro as well ? same for the others of this kind.